### PR TITLE
Laetitia week 3

### PR DESCRIPTION
**Pull request by supertata**
# Your Details

Your Name:
Your City:
Your Slack Name:

# Homework Details

Module:
Week:

### CODE REVIEW

File: week-3/Homework/mandatory/1-oxygen-levels.js [hunk 1]
```diff
@@ -27,9 +27,17 @@ console.log(newdessert); // returns ice cream
...
 
-function safeLevels() {
+const oxygenLevels = ["24.2%", "11.3%", "19.9%", "23.1%", "29.3%", "20.2%"]
 
+function safeLevels(arr) {
+for (let i = 0; i < arr.length; i++)
+ if (arr[i] > "19.5%" && arr[i] < "23.5%") {
+  return arr[i]
 }
+
```
Comment (anthonytranDev) [inline week-3/Homework/mandatory/1-oxygen-levels.js:37]: Good job in understanding the question.

However this one requires you to use an **array method**, as part of this exercise.

Hint: use `filter()`

File: week-3/Homework/mandatory/2-bush-berries.js [hunk 2]
```diff
@@ -24,29 +24,54 @@ will have a remainder of 0 i.e. they are even numbers*/
...
+  
+if (notPink === true) 
+  {
+  return console.log("this is toxic")
+  
+ } 
+ else {
+   return console.log("safe to eat")
+   // expected output: safe to eat
+ }
```
Comment (anthonytranDev) [inline week-3/Homework/mandatory/2-bush-berries.js:53]: styling review

Hey! Just to make it easier for developers of you team in the future. Recommend using `Prettier` extensions - don't be so drawn to keeping questions looking the same. If you can improve on something, please do.

It'll help you in creating more beautiful code! 👓

File: week-3/Homework/mandatory/2-bush-berries.js [hunk 3]
```diff
@@ -24,29 +24,54 @@ will have a remainder of 0 i.e. they are even numbers*/
...
 
-  console.log(`${test_name}: ${status}`);
-}
+//   console.log(`${test_name}: ${status}`);
+// }
 
-test("bushChecker funtion works - case 1", bushChecker(bushBerryColours1) === "Toxic! Leave bush alone!")
-test("bushChecker funtion works - case 1", bushChecker(bushBerryColours2) === "Bush is safe to eat from")
+// test("bushChecker funtion works - case 1", bushChecker(bushBerryColours1) === "Toxic! Leave bush alone!")
+// test("bushChecker funtion works - case 1", bushChecker(bushBerryColours2) === "Bush is safe to eat from")
```
Comment (anthonytranDev) [inline week-3/Homework/mandatory/2-bush-berries.js:77]: Remember not to modify these test statements.

Even, if you haven't managed to figure it out. They are there to help figure how to pass the question 😎

Comment (anthonytranDev) [reply week-3/Homework/mandatory/2-bush-berries.js:77]: Plus, and I believe you know this.

The questions still needs completing

File: week-3/Homework/mandatory/3-space-colonies.js [hunk 4]
```diff
@@ -13,24 +13,44 @@
...
-  "Aldous",
-  "Button family",
-  "Jude",
-  "Carmichael",
-  "Bunny",
-  "Asimov",
-  "Oscar family",
-  "Avery family",
-  "Archer family",
-];
```
Comment (anthonytranDev) [inline week-3/Homework/mandatory/3-space-colonies.js:54]: Again don't modify the test case.

If you do decide to modify something, just to test it out. Please remember to unchange it before each commit!

File: week-3/Homework/mandatory/4-eligible-students.js [hunk 5]
```diff
@@ -22,40 +22,69 @@
...
+// // Examples accessing to array element through array methods
+// // Example 1
+// let moreThanTwoArrays = twoDimensionalArray.filter(embeddedArray => embeddedArray.length > 2);
+// console.log(moreThanTwoArrays);
+// // This filter only keeps embededd arrays that have more than 3 elements
+
+// // Example 2
+// let arrayLengths = twoDimensionalArray.map(embeddedArray => embeddedArray.length);
+// console.log(arrayLengths);
+// // This gives back how many elements of each array have
```
Comment (anthonytranDev) [inline week-3/Homework/mandatory/4-eligible-students.js:51]: Same review as question 2

File: week-3/Homework/mandatory/4-eligible-students.js [hunk 6]
```diff
@@ -72,72 +101,90 @@ function getEligibleStudents() {
...
-)
+// test("eligibleStudents2 function without eligible candidates",
+//   arraysEqual(getEligibleStudents2(deltaStudentGroup), ["amy"])
+// )
 
-test("eligibleStudents2 function works",
-  arraysEqual(getEligibleStudents2(alphaStudentGroup), ["Ahmed", "Adam"])
-)
+// test("eligibleStudents2 function with no candidates",
+//   arraysEqual(getEligibleStudents2([]), []))
```
Comment (anthonytranDev) [inline week-3/Homework/mandatory/4-eligible-students.js:188]: Same review as in question 2

File: week-3/Homework/mandatory/5-journey-planner.js [hunk 7]
```diff
@@ -119,113 +157,129 @@ function getLocationName() { }
...
-    "Angel",
-    "Tower Bridge",
-    "Greenwich",
-  ])
+function journeyPlanner(array, string) {
+  
+  let availableLocations = array.filter((transport) =>
+  transport.includes(string)
+  
 );
```
Comment (anthonytranDev) [inline week-3/Homework/mandatory/5-journey-planner.js:173]: Same review as question 2

**Comment by supertata** [comment]
> Good work getting in the homework on time. 👍
> 
> There are several comments I've made below.
> 
> Plus please notify me earlier, if you were unable to do any of the questions; than keeping quiet. There is no shame in spending extra time, even additional week or so to do the questions

Thanks for your feedback Anthony. It will be useful for my revisions in the next week. I have actually completed all the exercises so I'm not sure what exercises you are referring but thanks anyway for making yourself available for extra support. I always learn a lot from attending your extra sessions and even from your replies to my questions in our group :)

**Pull request closed without merging**